internal/runbits/requirements/requirements.go

```diff
@@ -400,6 +355,137 @@ func (r *RequirementOperation) ExecuteRequirementOperation(
 	return nil
 }
 
+func (r *RequirementOperation) solve(commitID strfmt.UUID, ns *model.Namespace) (
+	_ *runtime.Runtime, _ *model.BuildResult, _ *artifact.ArtifactChangeset, rerr error,
+) {
+	// Initialize runtime
+	var trigger target.Trigger
+	switch ns.Type() {
+	case model.NamespaceLanguage:
+		trigger = target.TriggerLanguage
+	case model.NamespacePlatform:
+		trigger = target.TriggerPlatform
+	default:
+		trigger = target.TriggerPackage
+	}
+
+	spinner := output.StartSpinner(r.Output, locale.T("progress_solve_preruntime"), constants.TerminalAnimationInterval)
+
+	defer func() {
+		if rerr != nil {
+			spinner.Stop(locale.T("progress_fail"))
+		} else {
+			spinner.Stop(locale.T("progress_success"))
+		}
+	}()
+
+	rtTarget := target.NewProjectTarget(r.Project, &commitID, trigger)
+	rt, err := runtime.New(rtTarget, r.Analytics, r.SvcModel, r.Auth, r.Config, r.Output)
+	if err != nil {
+		return nil, nil, nil, locale.WrapError(err, "err_packages_update_runtime_init", "Could not initialize runtime.")
+	}
+
+	setup := rt.Setup(&events.VoidHandler{})
+	buildResult, err := setup.Solve()
+	if err != nil {
+		return nil, nil, nil, errs.Wrap(err, "Solve failed")
+	}
+
+	// Get old buildplan
+	// We can't use the local store here; because it might not exist (ie. integrationt test, user cleaned cache, ..),
+	// but also there's no guarantee the old one is sequential to the current.
+	commit, err := model.GetCommit(commitID, r.Auth)
+	if err != nil {
+		return nil, nil, nil, errs.Wrap(err, "Could not get commit")
+	}
+
+	var oldBuildPlan *bpModel.Build
+	if commit.ParentCommitID != "" {
+		bp := model.NewBuildPlannerModel(r.Auth)
+		oldBuildResult, err := bp.FetchBuildResult(commit.ParentCommitID, rtTarget.Owner(), rtTarget.Name())
+		if err != nil {
+			return nil, nil, nil, errs.Wrap(err, "Failed to fetch build result")
+		}
+		oldBuildPlan = oldBuildResult.Build
+	}
+
+	changedArtifacts, err := buildplan.NewArtifactChangesetByBuildPlan(oldBuildPlan, buildResult.Build, false, false, r.Config, r.Auth)
+	if err != nil {
+		return nil, nil, nil, errs.Wrap(err, "Could not get changed artifacts")
+	}
+
+	return rt, buildResult, &changedArtifacts, nil
+}
+
+func (r *RequirementOperation) cveReport(requirementName, requirementVersion string, artifactChangeset artifact.ArtifactChangeset, operation bpModel.Operation, ns *model.Namespace) error {
+	if !r.Auth.Authenticated() || operation == bpModel.OperationRemoved {
+		return nil
+	}
+
+	reqNameAndVersion := requirementName
```

Comment:
Why initialize this as a new variable? It's only being used in the spinner output.